clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -426,6 +428,17 @@ private void handshakeFinished() throws IOException {
                 SSLSession session = sslEngine.getSession();
                 log.debug("SSL handshake completed successfully with peerHost '{}' peerPort {} peerPrincipal '{}' cipherSuite '{}'",
                         session.getPeerHost(), session.getPeerPort(), peerPrincipal(), session.getCipherSuite());
+                if (sslInformation == null) {
+                    String cipherSuiteName = session.getCipherSuite();
+                    if (cipherSuiteName == null || cipherSuiteName.isEmpty()) {
+                        cipherSuiteName = "unknown";
+                    }
+                    String protocolName = session.getProtocol();
+                    if (protocolName == null || protocolName.isEmpty()) {
+                        protocolName = "unknown";
+                    }
+                    sslInformation = new SslInformation(cipherSuiteName, protocolName);
```

Comment:
@rajinisivaram Can you please double check if these methods can actually return `null`? It looked to me like they would not, but I didn't look deeply.

Reply:
From the default implementation and netty implementation, it looks like they would never be null. But I guess it is ok to do the null-check here since we may make SSLEngines pluggable.